### PR TITLE
Add custom data support

### PR DESCRIPTION
**Pull request by artememelin**
### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [ ] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,55 +53,51 @@ lwc_show(chart)
 
 ![chart](docs/src/assets/chart.png)
 
-Panel candlestick and histogram chart, (using data from [CryptoAPIs](https://github.com/bhftbootcamp/CryptoAPIs.jl)):
+Custom data types can also be visualized (e.g. time series from [TimeArrays](https://github.com/bhftbootcamp/TimeArrays.jl)). The following example shows how smoothed prices can be visualized:
```
Comment (dmitrii-doronin) [inline README.md]:
```suggestion
The package also support custom data types. In the following example we used [TimeArrays](https://github.com/bhftbootcamp/TimeArrays.jl) to smooth prices:
```

File: docs/src/index.md [hunk 2]
```diff
@@ -43,55 +43,51 @@ lwc_show(chart)
 
 ![chart](assets/chart.png)
 
-Panel candlestick and histogram chart, (using data from [CryptoAPIs](https://github.com/bhftbootcamp/CryptoAPIs.jl)):
+Custom data types can also be visualized (e.g. time series from [TimeArrays](https://github.com/bhftbootcamp/TimeArrays.jl)). The following example shows how smoothed prices can be visualized:
```
Comment (dmitrii-doronin) [inline docs/src/index.md]:
```suggestion
The package also support custom data types. In the following example we used [TimeArrays](https://github.com/bhftbootcamp/TimeArrays.jl) to smooth prices:
```

File: src/charts/baseline.jl [hunk 3]
```diff
@@ -32,15 +32,20 @@ struct BaseLineChartSettings <: AbstractChartSettings
...
+    lwc_baseline(custom_data::Vector; kw...) -> LWCChart
 
 Creates a [`LWCChart`](@ref) object that contains baseline chart information.
 The `timestamps` can be passed as `Vector{Integer}` of Unix time or `Vector{TimeType}`.
 You can also use type [`LWCSimpleChartData`](@ref) for more flexible color settings.
 
 Wrapper function for [`Baseline`](https://tradingview.github.io/lightweight-charts/docs/series-types#baseline).
 
+!!! note
+    You can use a `custom_data` with custom type elements for which is defined a [conversion method](https://docs.julialang.org/en/v1/base/base/#Base.convert) to type `Tuple` with two elements: `timestamp::Union{TimeType,Real}` and `value::Real`.
```
Comment (dmitrii-doronin) [inline src/charts/baseline.jl]:
```suggestion
    You can use a `custom_data` with custom type elements that have a [conversion method](https://docs.julialang.org/en/v1/base/base/#Base.convert) to type `Tuple` with two elements: `timestamp::Union{TimeType,Real}` and `value::Real`.
```

Comment (dmitrii-doronin) [reply src/charts/baseline.jl]: Fix this in other places, too, please

**Comment by artememelin** [comment]
```julia
using Dates
using LightweightCharts

lwc_line(
    1:100,
    1:100;
) |> lwc_show

lwc_line(
    1:100;
) |> lwc_show

lwc_line(
    map(x -> (DateTime(x), x), 1:100);
) |> lwc_show

# ================== #

struct MyStruct1
    timestamp::DateTime
    value::Float64
end

Base.convert(::Type{Tuple}, x::MyStruct1) = (x.timestamp, x.value)

lwc_line(
    map(x -> MyStruct1(DateTime(x), x), 1:100);
) |> lwc_show

# ================== #

struct MyStruct2
    timestamp::DateTime
    value::Float64
end

Base.convert(::Type{LWCSimpleChartItem}, x::MyStruct2) = LWCSimpleChartItem(x.timestamp, x.value; color = LightweightCharts.randcolor())

lwc_line(
    map(x -> MyStruct2(DateTime(x), x), 1:100);
) |> lwc_show

# ================== #

struct MyStruct3
    timestamp::DateTime
    name::String
    value::Float64
    size::Int64
end

Base.convert(::Type{Tuple}, x::MyStruct3) = (x.timestamp, x.name, x.value, x.size)

Base.convert(::Type{LWCSimpleChartItem}, x::MyStruct3) = LWCSimpleChartItem(x.timestamp, x.value)

lwc_line(
    map(x -> MyStruct3(DateTime(x), "a", x, 1), 1:100);
) |> lwc_show

# ================== #

struct MyStruct4
    timestamp::DateTime
    open::Float64
    high::Float64
    low::Float64
    close::Float64
end

Base.convert(::Type{LWCSimpleChartItem}, x::MyStruct4) = LWCSimpleChartItem(x.timestamp, (x.high + x.low) / 2; color = x.open > x.close ? "green" : "red")
Base.convert(::Type{LWCCandleChartItem}, x::MyStruct4) = LWCCandleChartItem(x.timestamp, x.open, x.high, x.low, x.close)

Base.convert(::Type{Tuple}, x::MyStruct4) = (x.timestamp, x.open, x.high, x.low, x.close)

lwc_line(
    map(x -> MyStruct4(DateTime(x), rand(), rand(), rand(), rand()), 1:100);
) |> lwc_show

lwc_candlestick(
    map(x -> MyStruct4(DateTime(x), rand(), rand(), rand(), rand()), 1:100);
) |> lwc_show
```

**Comment by gryumov** [comment]
@DeevsDeevs, how do you feel about the PR? We had to tweak the code a bit to make it work with libraries like TimeArrays.

**Review by gryumov — APPROVED** [review]

**Pull request merged**